Update Go version in CircleCI

Because currently the CircleCI environment is built on Go 1.9, using any Go 1.10+ features results in the build failing:

```
GOBIN=/go/src/github.com/Azure/draft/bin go install 

# github.com/Azure/draft/pkg/cmdline
pkg/cmdline/cmdline.go:180:16: undefined: strings.Builder
Makefile:28: recipe for target 'build' failed
make: *** [build] Error 2
Exited with code 2
```

This currently affects #871 

We should update the CircleCI environment to the latest Go release and this issue tracks the progress for that.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,11 +3,10 @@ jobs:
   build:
     working_directory: /go/src/github.com/Azure/draft
     docker:
-      - image: golang:1.9
+      - image: golang:1.11.1
     environment:
       AZURE_CONTAINER: "draft"
       AZURE_STORAGE_ACCOUNT: "azuredraft"
-      DOCKER_USERNAME: "bacongobbler"
     steps:
       - checkout
       - run: echo 'export PATH=/go/src/github.com/Azure/draft/bin:$PATH' >> $BASH_ENV

